### ISSUE TITLE
Cache Clearance Causing 404 Havoc

### ISSUE THREAD

**Issue by suning-li** [goharbor/harbor#20417]
Hello Harbor Team,

I've encountered a recurring issue with image caching. Specifically, after the cache is cleared, any requests to access images return a 404 error until the cache is eventually repopulated. This issue resolves itself after some time, but it does disrupt the normal flow of operations.

Steps to Reproduce:

GET https://xxxxx.com/v2/etscache/library/nextcloud/manifests/latest: NOT_FOUND: artifact etscache/library/nextcloud:latest not found

Wait about half an hour, get it again, and you can download it

Expected Behavior:
I would expect that there would be a mechanism to either prevent this issue or handle it more gracefully, rather than directly returning a 404 error to the client.

Additional Information:
- Harbor version: 2.7

I understand that this might be due to the cache rebuilding process, but any guidance on how to mitigate this issue or improve the experience would be greatly appreciated.

Thank you for your assistance and the great work you do!

Best regards,
Suning

**Comment by wy65701436** [comment]
when you say cache, do you mean the cache layer mentioned in the harbor yaml? https://github.com/goharbor/harbor/blob/main/make/harbor.yml.tmpl#L300

**Comment by suning-li** [comment]
You mentioned the Settings, we did not make any changes, the only difference is
We are using create a project as a proxy to get all the official file packages. As shown in the figure。
![WeChat2fe049ef05f68a5744c77ef5ab574dcd](https://github.com/goharbor/harbor/assets/109640552/bd302c20-71fd-4cea-b837-3d10a5010453)

**Comment by yexia553** [comment]
Hi harbor team,
I am the same team with suning-li, our issue is that we created an proxy cache to https://hub.docker.com. and our internal developer will pull imgaes from it. but we found that we get below error sometimes:
```bash
GET https://xxxxx.com/v2/xxx/library/nextcloud/manifests/latest: NOT_FOUND: artifact etscache/library/nextcloud:latest not found
```
we will could download it just wait for a while without any other action. I guess it related with proxy cache mechanism, but I didn;t find any documents about it in [https://goharbor.io/docs](https://goharbor.io/docs)

This issue does not only occur with the Nextcloud image; it happens with any image.

thank you
